addons/external-dns/gcp/main.tf

```diff
@@ -0,0 +1,26 @@
+data "terraform_remote_state" "infra" {
+  backend = "local"
+  config = {
+    path = "../../../infra/gcp/terraform.tfstate.d/gcp-${var.cluster_id}-${var.gcp_k8s_regions[var.cluster_id]}/terraform.tfstate"
+  }
+}
+
+data "terraform_remote_state" "dns" {
+  backend = "local"
+  config = {
+    path = "credentials/terraform.tfstate"
+  }
+}
+
+module "external_dns" {
+  source                     = "../../../modules/addons/external-dns"
+  cluster_name               = data.terraform_remote_state.infra.outputs.cluster_name
+  k8s_host                   = data.terraform_remote_state.infra.outputs.host
+  k8s_cluster_ca_certificate = data.terraform_remote_state.infra.outputs.cluster_ca_certificate
+  k8s_client_token           = data.terraform_remote_state.infra.outputs.token
+  dns_provider               = "google"
+  tsb_fqdn                   = var.tsb_fqdn
+  dns_zone                   = replace(var.tsb_fqdn, "/^[^\\.]+\\./", "")  # Remove the first part of the fqdn
```

Comment:
we must create a sub-zone per each cluster, i.e. the provided dns_zone should host a subzone thats with the cluster_name, i.e. $var.cluster_name-$var.dns_zone so it will keep the provided zones clean and manageable, especially if the public zones are being used

Reply:
Already replied to this previously.

Reply:
accounting we follow `module/gcp/external-dns ` with `sub-zone` approach.